API.yaml

```diff
@@ -1395,6 +1397,26 @@ paths:
                       type: array
                       items:
                         type: string
+  /health:
+    get:
+      summary: Get the health status of this server
+      tags:
+      - Diagnostic
+      responses:
+        200:
+          description: Should return a status describing whether the server is ready to receive requests or not
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  status: object
+                    description: The status of the server
+                    type: string
+                    enum:
+                    - "UP"
```

Comment:
Shouldn't we use HTTP codes to get the status? Unless we need to give other information that cannot be encoded in the HTTP error codes. When no server is running, the request will "Connection refused", which means down.

Reply:
Matthew suggested that too, and I'm not sure. As Simon mentioned, many APIs also have this kind of health check request which can return various status. It's more flexible as we can have more replies eventually. Also, HTTP code description says about 500: "the server failed to fulfil an apparently valid request". In this case, if the server bothers to give a down or unknown status, it did fulfill the request correctly. Of course, in our current case, it would reply 1 every time and will refuse connection if down

Reply:
UP and DOWN as status should be enough. Not sure when to use UNKOWN and what that would mean (UP or DOWN). I would remove UNKNOWN. Alternatively, we could have status "success" and "failure", but I don't have any strong opinion for either. Other HTTP return codes than 200 should be seen as failure. This needs to be also documented in this section.
Later on, it could return a "details" structure to give more details on the failure. Not for now.

Reply:
Usually the UNKNOW is mapped to a 503 error (service not available), because you can not expose a reason since the server is down. DOWN (can also be mapped to 503) is useful if this endpoint is detached from the trace server. That means the server is still accessible but not the "service" in this case you might be able to provide a cause.

Reply:
iiuc, a http code of 200 would come with a status of UP and http code 503 could come with a status of DOWN and optional something elses, so http code and/or status can be used by clients to know the real status.